lib/utils.py

```diff
@@ -78,8 +79,9 @@ def get_node_ip():
         that is being used as the default gateway.
         This should work on both MacOS X and Linux."""
         try:
-            default_interface = grep(netstat('-nr'), '-e', '^default', '-e' '^0.0.0.0').split()[-1]
-            my_ip = ifaddresses(default_interface)[2][0]['addr']
+            interface_id = next(iter([gateways()][0]['default']), None)
+            default_interface = [gateways()][0]['default'][interface_id][1]
```

Comment:
If I understand correctly 82 and 83 lines can be replaces by
```
default_interface = gateways()['default']
```

Reply:
This is not the same sadly.

A gateways()['default'] will return multiple interfaces and also gateway ips.

Reply:
oh, my bad, you're right :) I meant something like this:
```
gateways()['default'].items()[0][1][1]
```
I'm not saying that this is the best option, it just seems to me 82 and 83 lines is not the most elegant solution and its really hard to read

Reply:
Yes, it would work. But only in python2, in case that would be a problem. An easier to read solution that would also work in python3 would be:
`interface_id = list(gateways()['default'])[0]`

I dont really have a preferred solution here.